app/services/migration/parity_check/client.rb

```diff
@@ -70,8 +80,16 @@ def headers
       }
     end
 
-    def url(app:, path:)
-      Rails.application.config.npq_separation[:parity_check]["#{app}_url".to_sym] + path
+    def url(app:)
+      Rails.application.config.npq_separation[:parity_check]["#{app}_url".to_sym] + formatted_path
+    end
+
+    def formatted_path
+      @formatted_path ||= begin
+        return path unless options[:id] && path.include?(":id")
+
+        path.sub(":id", eval(options[:id]).to_s) # rubocop:disable Security/Eval
```

Comment:
Instead of doing eval from the yaml file content, can we pass in instructions instead?
eg:
```yaml
  "/api/v1/npq-applications/:id":
    id: 'random_lead_provider_ecf_id`
```

And then in `Client` have method to translate to action:
```ruby
def process_instruct(n)
  case n
  when "random_lead_provider_ecf_id"
    lead_provider.applications.order("RANDOM()").limit(1).pick(:ecf_id)'
  end
end

ecf_id = process_instruct(options[:id])
```

Reply:
![](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExN28yNGgydnJzcjRsOTFybHBraWd4cTdtczN3ODBtNXEyd3g2Y21hZyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/dbd6jN0Atb9i8/giphy.gif)

Reply:
hmm somehow prefer it in the yaml file 🤔 otherwise we'll have a huge list in the client

Reply:
I prefer it out of the YAML file; easier to test, avoids `eval` and less duplication